flow_action_components/ExecuteSOQLQuery/force-app/main/default/classes/ExecuteSOQL.cls

```diff
@@ -29,53 +22,66 @@ public with sharing class ExecuteSOQL {
                 soqlQuery = soqlQuery.substring(0, soqlQuery.indexOf(curClause));
             }
         }
-
         if (soqlQuery != null && soqlQuery.contains(' from ') && soqlQuery.contains('select ') && soqlQuery.contains(' where ')) {
-            Pattern mPattern = pattern.compile('(?<=from )(.*)(?= where)');
+            Pattern mPattern = pattern.compile('(?<=from )(.*)(?= where .+(\\(select .+\\)))');
             Matcher mMatcher = mPattern.matcher(soqlQuery);
-            mMatcher.find();
-            String sObjectType = mMatcher.group(0);
+            Boolean found = mMatcher.find();
+            String sObjectType = '';
+            if (found) {
+                sObjectType = mMatcher.group(0);
+                // Recurse through sub-queries
+                Pattern subPattern = pattern.compile('\\(select .+\\)');
+                Matcher subMatcher = subPattern.matcher(soqlQuery);
+                while (subMatcher.find()) {
+                    String subQuery = subMatcher.group(0).removeStart('(').removeEnd(')');
+                    String formattedSubQuery = replaceWithFormattedValues(subQuery);
+                   	soqlQuery = soqlQuery.replace(subQuery, formattedSubQuery);
+                }
+            } else {
+                mPattern = pattern.compile('(?<=from )(.*)(?= where)');            
+                mMatcher = mPattern.matcher(soqlQuery);
+                if (mMatcher.find()) {
+                    sObjectType = mMatcher.group(0);
+                } else {
+                    throw new ExecuteSOQLException('Unable to parse query string: ' + soqlQuery);
+                }
+            }
             Map<String, String> fieldNameValueMap = new Map<String, String>();
             List<String> fieldNames = new List<String>();
-
             mPattern = pattern.compile('(?<=where )(.*)');
             mMatcher = mPattern.matcher(soqlQuery);
             while (mMatcher.find()) {
-                String whereClause = mMatcher.group(0);
-                fieldNames.addAll(whereClause.split('\\(|\\)|>=|<=|!=|=|>|<| in\\(| like | in:| or| and'));
+                String whereClause = mMatcher.group(0).replaceAll('\\(select .+\\)', '(SUBQUERY)');
+                fieldNames.addAll(whereClause.split('\\(|\\)|>=|<=|!=|=|>|<| in\\(| not in\\(| like | in:| or| and'));
             }
-
             if (!fieldNames.isEmpty()) {
-                for (Integer i = fieldNames.size() - 1; i >= 0; i -= 2) {
+               for (Integer i = fieldNames.size() - 1; i >= 1; i -= 2) {
```

Comment:
For good measure. If an unhandled query string makes it to here, you would get an out of bounds error if you let `i` go to 0.